### PR TITLE
Add tic-tac-toe

### PR DESCRIPTION
**Pull request by mRmetr0**
Let me know if the code is clean enough

### CODE REVIEW

File: src/extensions/ttt.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        ctx: discord.ApplicationContext,
+        tile: int = discord.Option(
+            int, autocomplete=discord.utils.basic_autocomplete(
+            [i for i in range(9)])),
+        reset: bool = False,
+    ):
+        if reset or self.gameover:
+            self.__reset_board()
+            self.message = await ctx.respond(embed=self.__create_embed(f"Starting new game. \nIt's {self.currentplayer}'s turn", True))
+        elif self.__select_tile(tile):
```
Comment (MythicalSora) [inline src/extensions/ttt.py:98]: Could you turn the three different game state booleans into a single enum?
And could you turn this into a Match-Case Statement?

File: src/extensions/ttt.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                return True
+        return False
+
+    def __check_tie(self):
+        for tile in self.board:
+            if self.__is_number(tile):
+                return False
+        return True
+
+    def __is_number(self, value):
```
Comment (MythicalSora) [inline src/extensions/ttt.py:56]: First of all, this should be using a more specific Exception. Second of all, this seems like a completely unnecessary function.
Python has ```str.isnumeric()``` which should suffice.

If you really must have this function, I'd prefer a ternary operator.

**Review by MythicalSora — CHANGES_REQUESTED** [review]
Please look at the comments I've added and fix them, that's all ^^

File: src/extensions/ttt.py (hunk 1 above)
Comment (mRmetr0) [reply src/extensions/ttt.py]: Changed state check into enum in latest commit.

File: src/extensions/ttt.py (hunk 2 above)
Comment (mRmetr0) [reply src/extensions/ttt.py]: Removed the method, replaced it with python's alternative in the latest commit

**Review by MythicalSora — APPROVED** [review]
Lgtm

**Pull request merged**